consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,656 @@
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # create a mock_wait_certificate that does nothing in check_valid
+        mock_wait_certificate = mock.Mock()
+        mock_wait_certificate.check_valid.return_value = None
+
+        # set new_mock_wait_certificate local_mean and duration
+        mock_wait_certificate.local_mean = 0.0
+        mock_wait_certificate.duration = 1.0
+
+        # create a new_fork_mock_wait_certificate with a higher duration time
+        new_fork_mock_wait_certificate = mock.Mock()
+        new_fork_mock_wait_certificate.check_valid.return_value = None
+        new_fork_mock_wait_certificate.local_mean = 0.0
+        new_fork_mock_wait_certificate.duration = 2.0
+
+        # set mock_utils.deserialize_wait_certificate
+        # to return a specific value for each fork_head that is used in
+        # poet_fork_resolver.compare()
+        # with cur_fork_head being deserialized first
+        mock_utils.deserialize_wait_certificate.side_effect = \
+            [mock_wait_certificate,
+             new_fork_mock_wait_certificate
+             ]
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_poet_enclave_module
+        mock_poet_enclave_module = mock.Mock()
+        mock_poet_enclave_module.return_value = \
+            mock_poet_enclave_factory.get_poet_enclave_module.return_value
+
+        # create mock_cur_fork_head
+        mock_cur_fork_header = \
+            mock.Mock(
+                identifier='0123456789abcdefedcba9876543210',
+                signer_pubkey='90834587139405781349807435098745',
+                previous_block_id='2',
+                header_signature='00112233445566778899aabbccddeeff')
+
+        # create mock_new_fork_head with same previous block id
+        mock_new_fork_header = \
+            mock.Mock(
+                identifier='0123456789abcdefedcba9876543211',
+                signer_pubkey='90834587139405781349807435098745',
+                previous_block_id='2',
+                header_signature='00112233445566778899aabbccddeeff')
+
+        fork_resolver = \
+            poet_fork_resolver.PoetForkResolver(
+                block_cache=mock_block_cache,
+                state_view_factory=mock_state_view_factory,
+                data_dir=self._temp_dir,
+                config_dir=self._temp_dir,
+                validator_id='validator_deadbeef')
+
+        # Subtest 1: when current fork head has the smaller wait duration
+        with mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                        'LOGGER') as mock_logger:
+            fork_resolver.compare_forks(
```

Comment:
You need to wrap the call to `fork_resolver.compare_forks(...)` with `self.assertFalse()` to verify that the function is returning the correct value to indicate that the current fork head should be used.